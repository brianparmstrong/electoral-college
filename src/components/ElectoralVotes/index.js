import { number, string } from 'prop-types';

const ElectoralVotes = ({
  party,
  popularVoteTotals,
  winnerTakeAllTotal
}) => (
  <div className='electoralVotesWrapper'>
    <p className='electoralVotesLabel'>Winner-Take-All:</p>
    <span className='electoralVotes' id={`${party}-wta`}>{winnerTakeAllTotal}</span>
    <p className='electoralVotesLabel'>Proportional:</p>
    <span className='electoralVotes' id={`${party}-prop`}>{popularVoteTotals}</span>
  </div>
);

ElectoralVotes.propTypes = {
  party: string.isRequired,
  popularVoteTotals: number.isRequired,
  winnerTakeAllTotal: number.isRequired
};

export default ElectoralVotes;
