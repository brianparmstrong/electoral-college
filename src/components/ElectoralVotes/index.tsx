import { IfcElectoralVotes } from '../../types';

const ElectoralVotes = (electoralVotes: IfcElectoralVotes) => (
  <div className="electoralVotesWrapper">
    <p className="electoralVotesLabel">Winner-Take-All:</p>
    <span className="electoralVotes" id={`${electoralVotes.party}-wta`}>
      {electoralVotes.winnerTakeAllTotal}
    </span>
    <p className="electoralVotesLabel">Proportional:</p>
    <span className="electoralVotes" id={`${electoralVotes.party}-prop`}>
      {electoralVotes.popularVoteTotals}
    </span>
  </div>
);

export default ElectoralVotes;
