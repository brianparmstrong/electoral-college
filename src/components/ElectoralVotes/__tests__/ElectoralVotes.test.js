import React from 'react';
import renderer from 'react-test-renderer';
import ElectoralVotes from '../';

const MOCK_VOTE_INFO = {
  party: 'ind',
  popularVoteTotals: 10,
  winnerTakeAllTotal: 0
};

it('renders correctly', () => {
  const electoralVotes = renderer
    .create(
      <ElectoralVotes {...MOCK_VOTE_INFO} />
    )
    .toJSON();
  expect(electoralVotes).toMatchSnapshot();
});
