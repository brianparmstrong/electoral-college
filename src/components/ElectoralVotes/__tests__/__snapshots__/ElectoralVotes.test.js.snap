// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="electoralVotesWrapper"
>
  <p
    className="electoralVotesLabel"
  >
    Winner-Take-All:
  </p>
  <span
    className="electoralVotes"
    id="ind-wta"
  >
    0
  </span>
  <p
    className="electoralVotesLabel"
  >
    Proportional:
  </p>
  <span
    className="electoralVotes"
    id="ind-prop"
  >
    10
  </span>
</div>
`;
